### PR TITLE
apply_noise(flux, flux_err)

### PR DESCRIPTION
**Pull request by hombit**
- Adds `apply_noise()` function
- Fixes a bug in `PhysicalModel.get_band_fluxes()` (`evaluate` vs `_evaluate`)
- Changes flux -> bandflux terminology over many docstrings and some functions
- Updates SN Ia test and NB to generate noise and fix other stuff

### CODE REVIEW

**Review by jeremykubica — APPROVED** [review]

File: src/tdastro/effects/effect_model.py
[the remark this answers is not in the document]
Comment (hombit) [reply src/tdastro/effects/effect_model.py:1]: Thank you, it is a rebasing artifact I missed!

File: src/tdastro/astro_utils/noise_model.py [hunk 1]
```diff
@@ -68,11 +68,34 @@ def poisson_flux_std(
...
 
     return np.sqrt(total_variance) * zp
+
+
+def apply_noise(flux, flux_err, rng=None):
+    """Apply Gaussian noise to a bandflux measurement.
+
+    Parameters
+    ----------
+    flux : ndarray of float
```
[the remark this answers is not in the document]
Comment (hombit) [reply src/tdastro/astro_utils/noise_model.py]: Fixed!

File: tests/tdastro/sources/test_snia.py [hunk 2]
```diff
@@ -28,30 +29,27 @@ def draw_single_random_sn(
...
+    dec = source.get_param(state, "dec")
+    obs_index = np.array(opsim.range_search(ra, dec, radius=1.75))
 
-        times = obs["time"]
-        phase_obs = times - t0
-        times = np.sort(times[(phase_obs > -20) & (phase_obs < 50)])
-        # Note that we don't have filter info yet.
+    # Update obs_index to only include observations within SN lifespan
+    phase_obs = opsim[opsim.colmap["time"]][obs_index] - t0
+    obs_index = obs_index[(phase_obs > -20) & (phase_obs < 50)]
```
Comment (mi-dai) [inline tests/tdastro/sources/test_snia.py:38]:
```suggestion
    obs_index = obs_index[(phase_obs > -20 * (1. + z)) & (phase_obs < 50 * (1. + z))]
```

**Review by mi-dai — COMMENTED** [review]
I suggested a minor change in the output times. For SALT models it's easier/safer to show rest-frame phase from -20 to 50 (without cutting out too many points for higher redshift). We should revisit this in the future if we want to show the whole opsim range (where SALT model may fail for out of range times)

**Pull request merged**